src/views/profilePage/EditProfile.jsx

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable react/jsx-no-bind */
+/* eslint-disable jsx-a11y/no-autofocus */
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  Container,
+  Row,
+  Col,
+  Image,
+  Button,
+  Form,
+  FormControl,
+  Alert
+} from 'react-bootstrap';
+import { validateUserProfile } from '../../utils/validator';
+import './profile.scss';
+import Footer from '../../components/footer/Index';
+import { editProfileRequest, uploadImage } from '../../redux/actions/profile';
+import Header from '../../components/header/Header';
+
+const EditProfile = (props) => {
+  const { editProfile, profile, history } = props;
+  const initialFormState = {
+    firstname: profile.firstname,
+    lastname: profile.lastname,
+    bio: profile.bio,
+  };
+
+  const [errors, setErrors] = useState([]);
+  const [image, setImage] = useState(null);
+  const [imageUrl, setImageUrl] = useState('https://res.cloudinary.com/djdsxql5q/image/upload/v1547914942/ireporter/ujz7ozprk8hkyszjju7d.jpg');
+
+  const renderErrors = () => errors.map(error => (
+    <Alert key={error} variant="danger">{error}</Alert>
+  ));
+
+  const [formData, setFormData] = useState(initialFormState);
+
+  const cancelChanges = () => props.history.push('/userprofile');
+
+  const handleChange = (event) => {
+    const { target } = event;
+
+    setFormData({
+      ...formData,
+      [target.name]: target.value
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const validationErrors = validateUserProfile(formData);
+    if (validationErrors.length > 0) {
+      return setErrors(validationErrors);
+    }
+    editProfile(formData);
+    props.history.push('/userprofile');
+  };
+
+
+  const onChangeImage = (event) => {
+    setImage(event.target.files[0]);
+    setImageUrl(URL.createObjectURL(event.target.files[0]));
+  };
+
+  const upload = async (event) => {
+    event.preventDefault();
+    const res = await props.uploadImage(image);
```

Comment:
@Dubby20 what's `res`?

Reply:
It's a shorthand for response but I'll update it to have response instead of res